futurex_openedx_extensions/dashboard/views.py

```diff
@@ -292,6 +293,34 @@ def get(self, request: Any, username: str, *args: Any, **kwargs: Any) -> JsonRes
         )
 
 
+class DataExportTasksView(
+    ListAPIView, UpdateAPIView, FXViewRoleInfoMixin
+):  # pylint: disable=too-many-ancestors
+    """View to get the list of learners for a course"""
+    serializer_class = serializers.DataExportTaskSerializer
+    permission_classes = [FXHasTenantCourseAccess]
+    pagination_class = DefaultPagination
+    fx_view_name = 'exported_files_data'
+    fx_default_read_only_roles = ['staff', 'instructor', 'data_researcher', 'org_course_creator_group']
+    fx_default_read_write_roles = ['staff', 'instructor', 'data_researcher', 'org_course_creator_group']
+    fx_allowed_write_methods = ['PUT']
+    fx_view_description = (
+        'api/fx/export/v1/tasks/: Get the list of user generated tasks, '
+        'api/fx/export/v1/tasks/<task_id>: Update task.'
+    )
+    ordering = ['-created_at']
+
+    def get_queryset(self) -> QuerySet:
+        """Get the list of user tasks"""
+        return DataExportTask.objects.filter(user=self.request.user)
+
+    def get_object(self) -> DataExportTask:
+        """Override to ensure that the user can only update their own tasks."""
+        task_id = self.kwargs.get('task_id')
+        task = get_object_or_404(DataExportTask, id=task_id, user=self.request.user)
+        return task
+
+
```

Comment:
disable `PUT` completely
```suggestion
    def put(self, request, *args, **kwargs):
        Response({"detail": "PUT method is not allowed."}, status=status.HTTP_405_METHOD_NOT_ALLOWED)


```